spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/data/GenericsHelpers.java

```diff
@@ -70,6 +72,19 @@ public static void assertEqualsSafe(Types.StructType struct, Record expected, Ro
     }
   }
 
+  public static void assertEqualsBatch(
+      Types.StructType struct,
+      Iterator<Record> expectedRecords,
+      ColumnarBatch batch,
+      Map<Integer, Object> idToConstant,
+      Integer numRowsAlreadyRead) {
```

Comment:
I'm looking into cleaning this up, I needed to pass the number of rows already read by the vectorized reader since that's needed to determine the overall row position